pkg/validate/raw/validate/source_spec_validator.go

```diff
@@ -295,17 +296,27 @@ func MissingSecretRef(sourceType v1beta1.SourceType, o client.Object) status.Err
 // InvalidGCPSAEmail reports that a RepoSync/RootSync Resource doesn't have the
 //
 //	correct gcp service account suffix.
-func InvalidGCPSAEmail(sourceType v1beta1.SourceType, o client.Object) status.Error {
+func InvalidGCPSAEmail(sourceType configsync.SourceType, o client.Object) status.Error {
 	kind := o.GetObjectKind().GroupVersionKind().Kind
 	return invalidSyncBuilder.
 		Sprintf("%ss which specify spec.%s.auth as %q must use suffix <gcp_serviceaccount_name>.[%s]",
 			kind, sourceType, configsync.AuthGCPServiceAccount, gcpSASuffix).
 		BuildWithResources(o)
 }
 
+// IllegalGCPSAEmail reports that a RootSync/RepoSync declares an auth type that
+// doesn't allow gcpServiceAccountEmail to be declared.
+func IllegalGCPSAEmail(sourceType configsync.SourceType, o client.Object) status.Error {
+	kind := o.GetObjectKind().GroupVersionKind().Kind
+	return invalidSyncBuilder.
+		Sprintf("%ss that spec.%s.auth is not %q must not specify spec.%s.gcpServiceAccountEmail",
```

Comment:
I think this may be clearer without a double negative. For example, `%ss must only specify spec.%s.gcpServiceAccountEmail when using the %s auth type`

Reply:
SG. done